PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,438 @@
+  parametrization2p1To2p25_ = file_prefiringparams_->Get<TF1>(paramName);
+  paramName = "L1prefiring_muonparam_2.25To2.4_" + dataeraMuon_;
+  parametrization2p25To2p4_ = file_prefiringparams_->Get<TF1>(paramName);
+
+  if (parametrization0p0To0p2_ == nullptr || parametrization0p2To0p3_ == nullptr ||
+      parametrization0p3To0p55_ == nullptr || parametrization0p55To0p83_ == nullptr ||
+      parametrization0p83To1p24_ == nullptr || parametrization1p24To1p4_ == nullptr ||
+      parametrization1p4To1p6_ == nullptr || parametrization1p6To1p8_ == nullptr ||
+      parametrization1p8To2p1_ == nullptr || parametrization2p1To2p25_ == nullptr ||
+      parametrization2p25To2p4_ == nullptr) {
+    missingInputMuon_ = true;
+    if (!skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Muon parametrization not found for at least one bin. All prefiring weights set to 0." << std::endl;
+  }
+
+  paramName = "L1prefiring_muonparam_HotSpot_" + dataeraMuon_;
+  parametrizationHotSpot_ = file_prefiringparams_->Get<TF1>(paramName);
+  file_prefiringparams_->Close();
+  if ((dataeraMuon_.find("2016") != std::string::npos) && parametrizationHotSpot_ == nullptr) {
+    missingInputMuon_ = true;
+    if (!skipwarnings_)
+      edm::LogWarning("L1PrefireWeightProducer")
+          << "Year is 2016 and no Muon parametrization is found for hot spot. All prefiring weights set to 0."
+          << std::endl;
+  }
+}
+
+L1PrefiringWeightProducer::~L1PrefiringWeightProducer() {}
+
+void L1PrefiringWeightProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+
+  //Photons
+  const std::vector<pat::Photon>& thePhotons = iEvent.get(photons_token_);
+
+  //Jets
+  const std::vector<pat::Jet>& theJets = iEvent.get(jets_token_);
+
+  //Muons
+  const std::vector<pat::Muon>& theMuons = iEvent.get(muon_token_);
+
+  //Probability for the event NOT to prefire, computed with the prefiring maps per object.
+  //Up and down values correspond to the resulting value when shifting up/down all prefiring rates in prefiring maps.
+  double nonPrefiringProba[3] = {1., 1., 1.};        //0: central, 1: up, 2: down
+  double nonPrefiringProbaJet[3] = {1., 1., 1.};     //0: central, 1: up, 2: down
+  double nonPrefiringProbaPhoton[3] = {1., 1., 1.};  //0: central, 1: up, 2: down
+  double nonPrefiringProbaMuon[3] = {1., 1., 1.};    //0: central, 1: up, 2: down
+
+  for (const auto fluct : {fluctuations::central, fluctuations::up, fluctuations::down}) {
+    if (!missingInputEcal_) {
+      for (const auto& photon : thePhotons) {
+        double pt_gam = photon.pt();
+        double eta_gam = photon.eta();
+        if (pt_gam < 20.)
+          continue;
+        if (fabs(eta_gam) < 2.)
+          continue;
+        if (fabs(eta_gam) > 3.)
+          continue;
+        double prefiringprob_gam = getPrefiringRateEcal(eta_gam, pt_gam, h_prefmap_photon_, fluct);
+        nonPrefiringProba[fluct] *= (1. - prefiringprob_gam);
```

Comment:
`nonPrefiringProba[fluct]` is nothing else than the product of `nonPrefiringProbaPhoton[fluct]`,  `nonPrefiringProbaJet[fluct]` and `nonPrefiringProbaMuon[fluct]`.
As such, it could be more convenient to compute it only once before current [L322](https://github.com/cms-sw/cmssw/pull/33758/files#diff-4052fd6f31ddfc7812747e0a524c682f5344bc807fd6f55b2228e6a7bf2731f5R322):
```C++
  for (const auto fluct : {fluctuations::central, fluctuations::up, fluctuations::down}) {
     nonPrefiringProba[fluct] = nonPrefiringProbaPhoton[fluct] * nonPrefiringProbaJet[fluct] * nonPrefiringProbaMuon[fluct];
  }
```

Reply:
I think that's a good idea, I'll implement it.